### PR TITLE
Support stake pools on genesis

### PR DESCRIPTION
**Pull request by kderme**
support stake pools and delegation to stake pools in the genesis file

### CODE REVIEW

File: shelley/chain-and-ledger/executable-spec/test/Tests.hs [hunk 1]
```diff
@@ -21,12 +21,12 @@ mainTests :: TestTree
...
-      minimalPropertyTests,
-      serializationTests,
-      stsTests,
-      unitTests,
+    [ -- addressTests,
+      -- cddlTests 5,
+      -- minimalPropertyTests,
+      -- serializationTests,
+      -- stsTests,
+      -- unitTests,
```
Comment (JaredCorduan) [inline shelley/chain-and-ledger/executable-spec/test/Tests.hs]: let's not remove these tests :)

**Review by JaredCorduan — COMMENTED** [review]
In general, we need to be very careful about this kind of approach. This is allowing us to directly manipulate the ledger state in a way that bypasses the rules. If we later make any changes to how, for example. pool registrations work, then we need to remember to update the logic here as well. Otherwise we can end up with very different behaviors, depending on if we used this feature or not.

File: shelley/chain-and-ledger/executable-spec/test/Tests.hs (hunk 1 above)
Comment (kderme) [reply shelley/chain-and-ledger/executable-spec/test/Tests.hs]: oops, fixed

**Comment by kderme** [comment]
> In general, we need to be very careful about this kind of approach. This is allowing us to directly manipulate the ledger state in a way that bypasses the rules. If we later make any changes to how, for example. pool registrations work, then we need to remember to update the logic here as well. Otherwise we can end up with very different behaviors, depending on if we used this feature or not.

As far as I know this is meant for testing. I quote @dcoutts from slack 
```
So we have another feature request from @disasm to support stake pools and delegation to stake pools in the genesis file.
This would require doing this TODO
         <*> pure emptyGenesisStaking --TODO 
which involves writing To/FromJSON instances for a dozen extra types.
Sam and the wallet folks want this so that they can set up a single-node cluster that acts as a stake pool, not a BFT node, so they can try more things more easily.
```

File: shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/Genesis.hs [hunk 2]
```diff
@@ -156,10 +157,24 @@ instance Crypto crypto => FromJSON (ShelleyGenesis crypto) where
                 <$> obj .: "genDelegs"
             )
         <*> obj .: "initialFunds"
-        <*> pure emptyGenesisStaking --TODO
+        <*> obj .:? "staking" .!= emptyGenesisStaking
```
Comment (kderme) [inline shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/Genesis.hs:150]: ensures backwards compatibility

**Comment by JaredCorduan** [comment]
> As far as I know this is meant for testing.

Ok, sounds good then. We should just keep in mind, then (possibly with some comments somewhere...) that if the logic in the STS rules changes, this setup will need to change accordingly.

**Review by JaredCorduan — APPROVED** [review]
:+1:

**Pull request merged**